ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh

```diff
@@ -0,0 +1,261 @@
+  # use registry credential for disconnected auth
+  grep "auths" "${SHARED_DIR}/install-config.yaml" > "${day2_pull_secret}"
+  #oc get secret -n openshift-config pull-secret -o jsonpath='{.data.\.dockerconfigjson}' | base64 -d > "${day2_pull_secret}"
+fi
+
+echo "Create node.iso for day2 worker nodes..."
+/tmp/oc adm node-image create --dir="${INSTALL_DIR}" -a "${day2_pull_secret}" --insecure=true
+
+# Patching the cluster_name again as the one set in the ipi-conf ref is using the ${UNIQUE_HASH} variable, and
+# we might exceed the maximum length for some entity names we define
+# (e.g., hostname, NFV-related interface names, etc...)
+CLUSTER_NAME=$(<"${SHARED_DIR}/cluster_name")
+
+gnu_arch=$(echo "$architecture" | sed 's/arm64/aarch64/;s/amd64/x86_64/;')
+case "${BOOT_MODE}" in
+"iso")
+  ### Copy the image to the auxiliary host
+  echo -e "\nCopying the day2 node ISO image into the bastion host..."
+  scp "${SSHOPTS[@]}" "${INSTALL_DIR}/node.iso" "root@${AUX_HOST}:/opt/html/${CLUSTER_NAME}.node.iso"
+  echo -e "\nMounting the ISO image in the hosts via virtual media and powering on the hosts..."
+  # shellcheck disable=SC2154
+  for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
+    # shellcheck disable=SC1090
+    . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+    if [[ "${name}" == *-a-* ]] && [ "${ADDITIONAL_WORKERS_DAY2}" == "true" ]; then
+      if [ "${transfer_protocol_type}" == "cifs" ]; then
+        IP_ADDRESS="$(dig +short "${AUX_HOST}")"
+        iso_path="${IP_ADDRESS}/isos/${CLUSTER_NAME}.node.iso"
+      else
+        # Assuming HTTP or HTTPS
+        iso_path="${transfer_protocol_type:-http}://${AUX_HOST}/${CLUSTER_NAME}.node.iso"
+      fi
+      mount_virtual_media "${host}" "${iso_path}"
+    fi
+  done
+
+  wait
+  if [ -f /tmp/virtual_media_mount_failed ]; then
+    echo "Failed to mount the ISO image in one or more hosts"
+    exit 1
+  fi
+;;
+"pxe")
+  ### Create pxe files
+  echo -e "\nCreating PXE files..."
+  oinst agent create pxe-files
+  ### Copy the image to the auxiliary host
+  echo -e "\nCopying the PXE files into the bastion host..."
+  scp "${SSHOPTS[@]}" "${INSTALL_DIR}"/boot-artifacts/agent.*-vmlinuz* \
+    "root@${AUX_HOST}:/opt/dnsmasq/tftpboot/${CLUSTER_NAME}/vmlinuz_${gnu_arch}"
+  scp "${SSHOPTS[@]}" "${INSTALL_DIR}"/boot-artifacts/agent.*-initrd* \
+    "root@${AUX_HOST}:/opt/dnsmasq/tftpboot/${CLUSTER_NAME}/initramfs_${gnu_arch}.img"
+  scp "${SSHOPTS[@]}" "${INSTALL_DIR}"/boot-artifacts/agent.*-rootfs* \
+    "root@${AUX_HOST}:/opt/html/${CLUSTER_NAME}/rootfs-${gnu_arch}.img"
+;;
+*)
+  echo "Unknown install mode: ${BOOT_MODE}"
+  exit 1
+esac
+
+EXPECTED_NODES=$(( $(get_ready_nodes_count) + ADDITIONAL_WORKERS ))
+
+# shellcheck disable=SC2154
+for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
```

Comment:
```suggestion
for bmhost in $(yq e -o=j -I=0 '.[] | select(.name='worker-a-*') "${SHARED_DIR}/hosts.yaml"); do
```

To avoid the if in each of the for and just loop through the additional workers.

Have we confirmed that all the other install scripts skip worker-a-* nodes?

Reply:
Let me check and update all loops again.
Day1 install script skip all worker-a-* nodes.